openassessment/data.py

```diff
@@ -993,15 +999,37 @@ def create_zip_with_attachments(cls, file, course_id, submission_files_data):
 
         with ZipFile(file, 'w') as zip_file:
             for file_data in submission_files_data:
-                file_content = (
-                    cls._download_file_by_key(file_data['key'])
-                    if file_data['type'] == cls.ATTACHMENT
-                    else file_data['content']
-                )
-
-                zip_file.writestr(file_data['file_path'], file_content)
-                csvwriter.writerow(file_data)
+                clone_file_data = copy.copy(file_data)
+                key = clone_file_data['key']
+                file_path = clone_file_data['file_path']
+                try:
+                    file_content = (
+                        cls._download_file_by_key(key)
+                        if clone_file_data['type'] == cls.ATTACHMENT
+                        else clone_file_data['content']
+                    )
+                except FileMissingException:
+                    clone_file_data['file_found'] = False
+                    # added a header to csv file to indicate that the file was found or not.
+                    # not sure if I should create a {file_path}.error.txt to indicate the file error more clearly.
+                    logger.warning(
+                        'File: \'{file_path}\' with key: \'{key}\' was missing.\n'
+                        'Full detail:'
+                        '\n\tCourse Id: {course_id}'
+                        '\n\tBlock Id: {block_id}'
+                        '\n\tStudent Id: {student_id}'
+                        '\n\tKey: {key}'
+                        '\n\tName: {name}'
+                        '\n\tType: {type}'
+                        '\n\tDescription: {description}'
+                        '\n\tFile Path: {file_path}'
+                        '\n\tFile Size: {size}'.format(**clone_file_data)
+                    )
+                else:
+                    clone_file_data['file_found'] = True
+                    zip_file.writestr(file_path, file_content)
 
+                csvwriter.writerow(clone_file_data)
```

Comment:
I think there's a cleaner way to do this without duplicating or writing to the source object. Since you're only adding `file_found` key:

```suggestion
                csvwriter.writerow(**file_data, 'file_found': fileFound)
```

Reply:
Will this work? I'm not super familiar with this function call, but it looks like you're mixing syntax

Reply:
specifically, wouldn't you need to wrap this in {}? or no?

Reply:
Yes, the correct syntax is to wrap it in {}. so it would be `csvwriter.writerow({**file_data, 'file_found': False})`